spec/features/friends/user_can_add_friends_spec.rb

```diff
@@ -36,7 +36,20 @@
     click_button "Add Friend"
 
     expect(page).to_not have_content("Friend Added Successfully")
-    expect(page).to have_content("Invalid Email Entered, Try Again")
+    expect(page).to have_content("Invalid entry, please try a different username")
+  end
+
+  it "I cannot add myself as a friend" do
+    user = create(:user, username: 'user')
+
+    allow_any_instance_of(ApplicationController).to receive(:current_user).and_return(user)
+
+    visit profile_friends_path
+    fill_in :username, with: "user"
+    click_button "Add Friend"
+
+    expect(page).to_not have_content("Friend Added Successfully")
+    expect(page).to have_content("Invalid entry, please try a different username")
   end
```

Comment:
Never really thought that this would be an issue - but I guess that's good on other apps for not having this issue